src/main/java/loanbook/logic/commands/EditCommand.java

```diff
@@ -184,7 +209,7 @@ public EditLoanDescriptor(EditLoanDescriptor toCopy) {
          * Returns true if at least one field is edited.
          */
         public boolean isAnyFieldEdited() {
-            return CollectionUtil.isAnyNonNull(name, phone, email, tags);
+            return CollectionUtil.isAnyNonNull(name, nric, phone, email, bike, rate, startTime, endTime, tags);
```

Comment:
In fact, it is possible for endTime to be null, as this is the case when a loan is in progress.

Reply:
This method is within the `EditLoanDescriptor` class. Which basically has a non-null field for any field that is going to be updated in the Loan. For example, I have an `EditLoanDescriptor` object whose members are all null except for the NRIC of "F1234567Z". What that means is that when the EditCommand executes, it'll only update the NRIC and no other field. There's no link to whether any data field such as `endTime` is required or optional in the Loan class over here.